evalml/pipelines/components/component_base.py

```diff
@@ -20,6 +20,10 @@ class ComponentBase(ABC, metaclass=ComponentBaseMeta):
     """
 
     _default_parameters = None
+    # --> this won't make custom components invalid for fast mode, but lets us opt into it.
+    # Given that fast mode isn't default behavior and we'll have proper warnings about not
+    # using it for custom components, (maybe we raise a warning if we find custom?), this makes the most sense
+    _can_be_used_for_fast_partial_dependence = True
```

Comment:
I just wanted to highlight this behavior: Any component defaults to being valid for fast mode, so if a user created a custom component or a developer added a new component, it's possible they might get incorrect PD results for pipelines containing it. 

I added a test in `test_components.py` that will raise an error when a new component is added to evalml, forcing developers to update the test and hopefully spend a little time thinking about if their component should be used for fast mode. It should be very quick to update, and devs can just default to disallowing it if they want. If people think that's a pain in the butt, I can remove it. 

There's nothing currently in place to stop custom components, but I was considering adding a warning using the `all_components` util to let users know the results may be wrong if a custom component is detected. Does anyone feel strongly about whether this warning should exist or not?

Reply:
I think its ok for now since fast mode is off by default but we should file an issue to track it!

Reply:
Created an issue: https://github.com/alteryx/evalml/issues/3817

While thinking about this more, it occurred to me that we could add a check in fast mode itself as to whether the pipeline relies on multiple columns. I think that check will be a little tricky to implement correctly, and we should test the impact on fast mode's runtime, so we shouldn't do that here, but I outlined the idea in the issue!